### ISSUE TITLE
PageChoosers are not filtered by locale on open

### ISSUE THREAD

**Issue by enzedonline** [wagtail/wagtail#11573]
### Issue Summary

There are a couple of issues happening here connected with this.
The first is that when opening a chooser with il8n enabled, the locale dropdown shows the current locale but the list is not filtered. To filter requires changing the locale. If you want the initially selected locale, ou have to change then change back again.
The second is that the default locale filter on opening is set to the active locale - this isn't related to the locale of the object being edited (ie the instance that the chooser is setting a field for). Active locale will only return the language the editor is using, not the language of the object instance.

This issue is not present with the SnippetChooser/ generic chooser. Locale filtering is handled somewhat differently.

### Steps to Reproduce

On a site with il8n enabled and multi-languages set up (my example: en, es, fr, de), to a page model, add a foreign key to  Page:
```python
    some_page = models.ForeignKey(
        "wagtailcore.Page",
        null=True,
        blank=True,
        related_name="+",
        on_delete=models.SET_NULL,
        verbose_name="Some Page",
    )
```
and add the corresponding FieldPanel.

With English activated as the current locale, create a new instance of your page model under the Spanish root.
Select a page. The chooser will have English as the selected locale and the chooser will show pages from all locales.

![image](https://github.com/wagtail/wagtail/assets/72310199/2a633aff-9904-460f-bf86-0d1da8f750ac)

This page list should be set to Spanish and show only Spanish pages.

The selected Locale issue seems to reside in BrowseView.get() (wagtail\admin\views\chooser.py):

```python
        if self.i18n_enabled:
            if self.parent_page.is_root():
                # 'locale' is the current value of the "Locale" selector in the UI
                if request.GET.get("locale"):        # <-- locale of object being edited is not passed in get request
                    selected_locale = get_object_or_404(
                        Locale, language_code=request.GET["locale"]
                    )
                    active_locale_id = selected_locale.pk
                else:
                    active_locale_id = Locale.get_active().pk                # <-- active locale is not relevant here

                # we are at the Root level, so get the locales from the current pages
                choose_url = reverse("wagtailadmin_choose_page")
                locale_options = [
                    {
                        "locale": locale,
                        "url": choose_url
                        + "?"
                        + urlencode(
                            {
                                "page_type": page_type_string,
                                "locale": locale.language_code,
                            }
                        ),
                    }
                    for locale in Locale.objects.filter(
                        pk__in=pages.values_list("locale_id")
                    ).exclude(pk=active_locale_id)
                ]
```

From what i can see, there is no reference to the calling object in BrowseView. The generic chooser `LocaleFilterMixin` handles this correctly by using `self.cleaned_data.get("locale")` (wagtail\admin\forms\choosers.py L147).

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

- The locale filtering should be activated on open. 
- The default selected locale should be the locale of the object instance the chooser is for, not the locale that the editor is browsing in  these two locales are not related.
- This issue does not exist for the Snippet chooser - the object locale is selected and listing is filtered correctly.

I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: Y

### Technical details

-   Python version: 3.12
-   Django version: L4.2.8
-   Wagtail version: 5.2.3
-   Browser version: Chrome 120

**Comment by abdlrhman08** [comment]
@enzedonline, I just read through `BrowserView.get()` fast and I don't believe the locale filtering problem is in it. I believe the problem is that the locale query parameter is not sent when you first open the chooser and to confirm that, if you change the locale in the locale selector it will function properly.

**Comment by abdlrhman08** [comment]
I believe the issue lies  [here](https://github.com/wagtail/wagtail/blob/025249aab78f6908e093fd94ff676317a20932fe/client/src/entrypoints/admin/page-chooser-modal.js#L239-L255). The locale parameter is not provided.

**Comment by enzedonline** [comment]
Great! That's the problem with the first part of the issue where the pages aren't initially filtered.
The second part of the issue where the initial language is being incorrectly picked in the dropdown is caused by `active_locale_id = Locale.get_active().pk`  in `BrowseView.get()`. The active locale has nothing to do with the object being edited.

**Comment by abdlrhman08** [comment]
@enzedonline, I was wrong in the first comment, my bad. The issue actually lies [here](https://github.com/wagtail/wagtail/blob/c384975147f4d8cfbb743761e507879f4205415b/wagtail/admin/views/chooser.py#L251-L253) 
```python
# Find parent page
if parent_page_id:
    self.parent_page = get_object_or_404(Page, id=parent_page_id)
elif self.desired_classes == (Page,):
    # Just use the root page
    self.parent_page = Page.get_first_root_node()
```
When the choose button is clicked, a GET request is made with the query parameter `page_type=wagtailcore.page` and as the if statements logic goes, the parent page will be defined as the root page and not the page where you have the chooser in.

>The second part of the issue where the initial language is being incorrectly picked in the dropdown is caused by `active_locale_id = Locale.get_active().pk` in `BrowseView.get()`. The active locale has nothing to do with the object being edited.

and as you stated above this leads the active locale to be the current active locale.

```python
if request.GET.get("locale"):
   selected_locale = get_object_or_404(
       Locale, language_code=request.GET["locale"]
    )
    active_locale_id = selected_locale.pk
else:
    active_locale_id = Locale.get_active().pk
    selected_locale = get_object_or_404(     #<--- I believe this is needed here
        Locale, id=active_locale_id
     )
````
I believe the commented line of code above is needed as they aren't filtered in the first place if the parent is the root. The whole issue would be fixed if the parent page could be correctly defined, because it will lead to [this](https://github.com/wagtail/wagtail/blob/c384975147f4d8cfbb743761e507879f4205415b/wagtail/admin/views/chooser.py#L319) line of code to be executed, which will filter the pages correctly. I just can't figure a way to set `self.parent_page` correctly

**Comment by enzedonline** [comment]
```
    active_locale_id = Locale.get_active().pk
    selected_locale = get_object_or_404(     #<--- I believe this is needed here
        Locale, id=active_locale_id
     )
```

A locale id is needed, but it needs to be the locale id of the instance, not the active locale id. The active locale id is just whatever language the editor is browsing in, it's not related to the object being edited. But even this doesn't seem to be working, it's just selecting the first in the list (or the default locale?).

Here, I'm editing a page in the DE locale with the editor's language set to Spanish:
![image](https://github.com/wagtail/wagtail/assets/72310199/ea96a3f5-b088-4b3b-ac96-0fc0cd8b483f)

If I open the page chooser, the locale selector is set to English (the default locale):
![image](https://github.com/wagtail/wagtail/assets/72310199/c8c8f96b-22ef-40a2-a930-98ffa3f02995)

The selector should be set to DE (the local of the instance, not the active or the default locale) then the filtering applied.

**Comment by enzedonline** [comment]
If I open a snippet chooser on the same page, the locale selector is correctly set to the instance locale and the filtering applied.

![image](https://github.com/wagtail/wagtail/assets/72310199/950cf845-5aff-4a0d-b0f0-3ea84d1bfee7)

This is how the page chooser should be functioning.

**Comment by abdlrhman08** [comment]
> A locale id is needed, but it needs to be the locale id of the instance, not the active locale id. 

Yes that's true, and the reason that this not happening is that when no `parent_page_id` is passed in the query parameters, the `self.parent_page` is defaulted to the root page in the database, this happens [here](https://github.com/wagtail/wagtail/blob/502dd7c723a3e0740915803b8fab6b9814bb9a9d/wagtail/admin/views/chooser.py#L249-L253)

> But even this doesn't seem to be working, it's just selecting the first in the list (or the default locale?).

The reason of this is that in [this](https://github.com/wagtail/wagtail/blob/502dd7c723a3e0740915803b8fab6b9814bb9a9d/wagtail/admin/views/chooser.py#L289-L297) if block, since the `self.parent_page` is set as the root page, the `active_locale_id` is set as the editors current locale, but still there is a missing part, the `selected_locale` variable is missing which is the one responsible for the filtering of the choices as it is shown in this if statement [here](https://github.com/wagtail/wagtail/blob/502dd7c723a3e0740915803b8fab6b9814bb9a9d/wagtail/admin/views/chooser.py#L346C13-L347C61), This block of code:
```python
selected_locale = get_object_or_404(
        Locale, id=active_locale_id
)
  ```
  when added after [this](https://github.com/wagtail/wagtail/blob/502dd7c723a3e0740915803b8fab6b9814bb9a9d/wagtail/admin/views/chooser.py#L297) line, fixes the second problem, but still the main problem remains unfixed, that the instances locale is not the one selected because the `self.parent_page` is incorrectly set to root. I am currently going through the code and trying to understand why does SnippetChooser work correctly. I will write back once I find anything

**Comment by enzedonline** [comment]
> the active_locale_id is set as the editors current `locale```

This is still not correct though - as I've been trying to point out, the editor's current locale has nothing to do with the locale that should be selected and filtered by initially.

In the example I posted above, Spanish is the editor's locale, but the object being edited is in the German locale. The selector should initialise with the object's locale, not the editor's locale. The editor's locale is not related to this - it's entirely irrelevant here.

**Comment by abdlrhman08** [comment]
> The selector should initialise with the object's locale, not the editor's locale. The editor's locale is not related to this - it's entirely irrelevant here.

Yes, I am aware of this, and currently I am trying to find a way to set `self.parent_page` correctly, which will lead to fixing this bug.

**Comment by laymonage** [comment]
I think this has been fixed in #13043, which will be released in 7.0. Please report back if the issue is still there, thanks!
